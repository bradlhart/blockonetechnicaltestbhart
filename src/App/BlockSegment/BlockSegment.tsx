import React, { useState } from 'react';

export const BlockSegment: React.FC = ({ blockInfo }) => {
  const [expanded, setExpanded] = useState(false);

  const expandRawData = () => setExpanded(!expanded);

  return (
    <div className="ui segment" onClick={expandRawData}>
      <div className="ui three column divided grid">
        <div className="ten wide column">{blockInfo.id}</div>
        <div className="three wide column">{blockInfo.timestamp}</div>
        <div className="three wide column">{blockInfo.transactions ? blockInfo.transactions.length : null}</div>
      </div>
      {expanded ? (
        <div className="ui segment" style={{ maxHeight: '200px', overflow: 'auto' }}>
          <pre>{JSON.stringify(blockInfo, null, 2)}</pre>
        </div>
      ) : null}
    </div>
  );
};
