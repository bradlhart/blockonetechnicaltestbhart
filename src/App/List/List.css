.listContent {
    word-break: break-word;
}