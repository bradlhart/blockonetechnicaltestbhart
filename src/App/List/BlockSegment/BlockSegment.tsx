import React, { useState } from 'react';
import { GetBlockResult } from 'eosjs/dist/eosjs-rpc-interfaces';
import './BlockSegment.css';

interface BlockSegmentProps {
  blockInfo: GetBlockResult;
}

export const BlockSegment: React.FC<BlockSegmentProps> = ({ blockInfo }: BlockSegmentProps) => {
  const [expanded, setExpanded] = useState<boolean>(false);

  const expandRawData = () => setExpanded(!expanded);

  return (
    <div className="ui segment" onClick={expandRawData}>
      <div className="ui three column divided grid listContent">
        <div className="seven wide mobile eight wide tablet ten wide computer column">{blockInfo.id}</div>
        <div className="five wide mobile five wide tablet four wide computer column">{blockInfo.timestamp}</div>
        <div className="four wide mobile three wide tablet two wide computer column">{blockInfo.transactions ? blockInfo.transactions.length : null}</div>
      </div>
      {expanded ? (
        <div className="ui segment rawDisplayContainer">
          <pre>{JSON.stringify(blockInfo, null, 2)}</pre>
        </div>
      ) : null}
    </div>
  );
};
