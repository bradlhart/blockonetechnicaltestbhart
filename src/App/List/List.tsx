import React, { ReactElement } from 'react';
import './List.css';

import { ListHeader } from './ListHeader/ListHeader';

interface ListProps {
  loading: boolean;
  errorMsg: string;
  blockList: ReactElement<any>[];
}

export const List: React.FC<ListProps> = ({ loading, errorMsg, blockList }: ListProps) => {
  return (
    <div className="ui segments">
      <ListHeader />
      {errorMsg ? (
        <div className="ui segment center aligned">{errorMsg}</div>
      ) : loading ? (
        <div className="ui segment center aligned">Fetching Data...</div>
      ) : blockList.length === 0 ? (
        <div className="ui segment center aligned">No Data To Display, Click Load To Retrieve Data</div>
      ) : (
        blockList
      )}
    </div>
  );
};
