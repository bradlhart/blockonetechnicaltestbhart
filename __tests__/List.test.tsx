import React from 'react';
import { shallow, ShallowWrapper } from "enzyme";

import List from '../src/App/List/List';
import BlockSegment from "../src/App/List/BlockSegment/BlockSegment";

const GetBlockResultMock = {
    timestamp: "timestamp",
    producer: "producer",
    confirmed: 0,
    previous: "previous",
    transaction_mroot: "transaction_mroot",
    action_mroot: "action_mroot",
    schedule_version: 0,
    producer_signature: "producer_signature",
    id: "id",
    block_num: 0,
    ref_block_prefix: 0,
};

describe('List', () => {
    let wrapper: ShallowWrapper<any>;
    const props = {
        loading: false,
        errorMsg: '',
        blockList: []
    };

    beforeEach(() => {
        wrapper = shallow(<List {...props} />);
    });

    test('renders correctly with default props', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('renders correctly with props in loading state', () => {
        wrapper.setProps({...props, loading: true});
        expect(wrapper).toMatchSnapshot();
    });

    test('renders correctly with props in error state', () => {
        wrapper.setProps({...props, errorMsg: 'Error Retrieving Chain Info:'});
        expect(wrapper).toMatchSnapshot();
    });

    test('renders correctly with props containing data', () => {
        wrapper.setProps({...props, blockList: [<BlockSegment key={GetBlockResultMock.id} blockInfo={GetBlockResultMock}/>]});
        expect(wrapper).toMatchSnapshot();
    });
});
