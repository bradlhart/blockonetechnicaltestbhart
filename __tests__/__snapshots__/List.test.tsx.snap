// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders correctly with default props 1`] = `
<div
  className="ui segments"
>
  <ListHeader />
  <div
    className="ui segment center aligned"
  >
    No Data To Display, Click Load To Retrieve Data
  </div>
</div>
`;

exports[`List renders correctly with props containing data 1`] = `
<div
  className="ui segments"
>
  <ListHeader />
  <BlockSegment
    blockInfo={
      Object {
        "action_mroot": "action_mroot",
        "block_num": 0,
        "confirmed": 0,
        "id": "id",
        "previous": "previous",
        "producer": "producer",
        "producer_signature": "producer_signature",
        "ref_block_prefix": 0,
        "schedule_version": 0,
        "timestamp": "timestamp",
        "transaction_mroot": "transaction_mroot",
      }
    }
    key="id"
  />
</div>
`;

exports[`List renders correctly with props in error state 1`] = `
<div
  className="ui segments"
>
  <ListHeader />
  <div
    className="ui segment center aligned"
  >
    Error Retrieving Chain Info:
  </div>
</div>
`;

exports[`List renders correctly with props in loading state 1`] = `
<div
  className="ui segments"
>
  <ListHeader />
  <div
    className="ui segment center aligned"
  >
    Fetching Data...
  </div>
</div>
`;
